### PR TITLE
Login up

### PR DESCRIPTION
**Pull request by molinajulian**
## Summary
Allowed create a user and create errors personalized to invalid passwords and emails.

## Trellocard
https://trello.com/c/r7SbJKNs

### CODE REVIEW

File: Training_node/first_project/app/controllers/users.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+'use strict';
```
Comment (gonziesc) [inline Training_node/first_project/app/controllers/users.js]: delete this

File: Training_node/first_project/app/controllers/users.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+const User = require('../models/').user,
+  validator = require('validator'),
+  bcrypt = require('bcryptjs'),
+  errors = require('../errors'),
+  logger = require('../logger');
+
+const validate = (email, password) => {
+  if (!(validator.isEmail(email) && validator.contains(email, '@wolox.com.ar'))) {
+    throw errors.emailInvalid('The email must be valid and pertain Wolox');
```
Comment (gonziesc) [inline Training_node/first_project/app/controllers/users.js]: don't throw errors, just return them

File: Training_node/first_project/app/controllers/users.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  const saltRounds = 10;
+  const user = req.body
+    ? {
+        firstName: req.body.firstName,
+        lastName: req.body.lastName,
+        password: req.body.password,
+        email: req.body.email
+      }
+    : {};
+  try {
```
Comment (gonziesc) [inline Training_node/first_project/app/controllers/users.js]: and do not try here

File: Training_node/first_project/app/controllers/users.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    validate(user.email, user.password);
+  } catch (e) {
+    return next(e);
+  }
+  bcrypt
```
Comment (gonziesc) [inline Training_node/first_project/app/controllers/users.js]: i would do this in a helper or service

File: Training_node/first_project/app/controllers/users.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  const saltRounds = 10;
+  const user = req.body
+    ? {
+        firstName: req.body.firstName,
+        lastName: req.body.lastName,
+        password: req.body.password,
+        email: req.body.email
+      }
+    : {};
+  let validation = validate(user.firstName, user.lastName, user.password, user.email);
```
Comment (gonziesc) [inline Training_node/first_project/app/controllers/users.js]: why let? it should be const

Comment (molinajulian) [reply Training_node/first_project/app/controllers/users.js]: @gonziesc because the variable validation is assigned again later

Comment (gonziesc) [reply Training_node/first_project/app/controllers/users.js]: do not use the same var then

**Pull request merged**